src/transformers/trainer.py

```diff
@@ -646,8 +671,10 @@ def train(self, model_path: Optional[str] = None, trial: Union["optuna.Trial", D
                     else True
                 ),
             )
-        # find_unused_parameters breaks checkpointing as per
-        # https://github.com/huggingface/transformers/pull/4659#issuecomment-643356021
+            # find_unused_parameters breaks checkpointing as per
+            # https://github.com/huggingface/transformers/pull/4659#issuecomment-643356021
+        elif self.sharded_dpp:
+            model = ShardedDDP(model, self.optimizer)
```

Comment:
Won't it be more readable to reverse these if/elif and result in an simpler to follow logic?
```
if self.sharded_dpp:
   ...
elif self.args.local_rank != -1:
   ...
```